sdk/core/azure-core-http-jdk11-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkAsyncHttpClientBuilder.java

```diff
@@ -0,0 +1,150 @@
+     */
+    public JdkAsyncHttpClientBuilder connectionTimeout(Duration connectionTimeout) {
+        // setConnectionTimeout can be null
+        this.connectionTimeout = connectionTimeout;
+        return this;
+    }
+
+    /**
+     * Sets the proxy.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.core.http.jdk.httpclient.JdkAsyncHttpClientBuilder.proxy#ProxyOptions}
+     *
+     * @param proxyOptions The proxy configuration to use.
+     * @return the updated {@link JdkAsyncHttpClientBuilder} object
+     */
+    public JdkAsyncHttpClientBuilder proxy(ProxyOptions proxyOptions) {
+        // proxyOptions can be null
+        this.proxyOptions = proxyOptions;
+        return this;
+    }
+
+    /**
+     * Sets the configuration store that is used during construction of the HTTP client.
+     * <p>
+     * The default configuration store is a clone of the {@link Configuration#getGlobalConfiguration() global
+     * configuration store}, use {@link Configuration#NONE} to bypass using configuration settings during construction.
+     *
+     * @param configuration The configuration store used to
+     * @return The updated JdkAsyncHttpClientBuilder object.
+     */
+    public JdkAsyncHttpClientBuilder configuration(Configuration configuration) {
+        this.configuration = configuration;
+        return this;
+    }
+
+    /**
+     * Build a HttpClient with current configurations.
+     *
+     * @return a {@link HttpClient}.
+     */
+    public HttpClient build() {
+        java.net.http.HttpClient.Builder httpClientBuilder = this.httpClientBuilder == null
+                     ? java.net.http.HttpClient.newBuilder()
+                     : this.httpClientBuilder;
+
+        httpClientBuilder = (this.connectionTimeout != null)
+            ? httpClientBuilder.connectTimeout(this.connectionTimeout)
+            : httpClientBuilder.connectTimeout(DEFAULT_CONNECT_TIMEOUT);
+
+        Configuration buildConfiguration = (configuration == null)
+            ? Configuration.getGlobalConfiguration()
+            : configuration;
+
+        ProxyOptions buildProxyOptions = (proxyOptions == null && buildConfiguration != Configuration.NONE)
+            ? ProxyOptions.fromConfiguration(buildConfiguration)
+            : proxyOptions;
+
+        if (executor != null) {
+            httpClientBuilder.executor(executor);
+        }
+
+        if (buildProxyOptions != null) {
+            httpClientBuilder = httpClientBuilder.proxy(new JdkHttpClientProxySelector(
+                buildProxyOptions.getType().toProxyType(),
+                buildProxyOptions.getAddress(),
+                buildProxyOptions.getNonProxyHosts()));
+
+            if (buildProxyOptions.getUsername() != null) {
+                httpClientBuilder.authenticator(new Authenticator() {
+                    @Override
+                    protected PasswordAuthentication getPasswordAuthentication() {
```

Comment:
Would this handle digest authentication?

Reply:
good question :) it won't. Is our current digest auth code just pluggable? or it's tailored per HTTP client? I haven't looked into that yet.

Reply:
digest-auth is on hold for now, like we discussed this will be picked up once we hear from native HTTP client owner.